pallets/asset-registry/src/lib.rs

```diff
@@ -615,27 +668,90 @@ impl<T: Config> GetByKey<T::AssetId, T::Balance> for Pallet<T> {
 pub struct XcmRateLimitsInRegistry<T>(PhantomData<T>);
 /// Allows querying the XCM rate limit for an asset by its id.
 /// Both a unknown asset and an unset rate limit will return `None`.
-impl<T: Config> GetByKey<T::AssetId, Option<T::Balance>> for XcmRateLimitsInRegistry<T> {
-	fn get(k: &T::AssetId) -> Option<T::Balance> {
+impl<T: Config> GetByKey<T::AssetId, Option<Balance>> for XcmRateLimitsInRegistry<T> {
+	fn get(k: &T::AssetId) -> Option<Balance> {
 		Pallet::<T>::assets(k).and_then(|details| details.xcm_rate_limit)
 	}
 }
 
-impl<T: Config> CreateRegistry<T::AssetId, T::Balance> for Pallet<T> {
+impl<T: Config> CreateRegistry<T::AssetId, Balance> for Pallet<T> {
 	type Error = DispatchError;
 
-	fn create_asset(name: &[u8], kind: AssetKind, existential_deposit: T::Balance) -> Result<T::AssetId, Self::Error> {
-		let bounded_name: BoundedVec<u8, T::StringLimit> = Self::to_bounded_name(name.to_vec())?;
-		Pallet::<T>::register_asset(bounded_name, kind.into(), existential_deposit, None, None)
+	fn create_asset(name: &[u8], kind: AssetKind, existential_deposit: Balance) -> Result<T::AssetId, Self::Error> {
+		let bounded_name = Some(Self::to_bounded_name(name.to_vec())?);
+
+		Pallet::<T>::do_register_asset(
+			None,
+			&AssetDetails::new(bounded_name, kind.into(), existential_deposit, None, None, None, false),
+			None,
+		)
+	}
+}
+
+impl<T: Config> Inspect<T::AssetNativeLocation, Balance> for Pallet<T> {
+	type Error = DispatchError;
+	type AssetId = T::AssetId;
+
+	fn is_sufficient(id: Self::AssetId) -> bool {
+		match Self::assets(id) {
+			Some(a) => a.is_sufficient,
+			None => false,
+		}
+	}
+}
+
+impl<T: Config> Mutate<T::AssetNativeLocation, Balance> for Pallet<T> {
+	fn set_location(asset_id: Self::AssetId, location: T::AssetNativeLocation) -> Result<(), Self::Error> {
+		Self::do_set_location(asset_id, location)
+	}
+}
+
+impl<T: Config> Create<T::AssetNativeLocation, Balance> for Pallet<T> {
+	fn register_asset(
+		asset_id: Option<Self::AssetId>,
+		name: Option<&[u8]>,
+		kind: AssetKind,
+		existential_deposit: Option<Balance>,
+		symbol: Option<&[u8]>,
+		decimals: Option<u8>,
+		location: Option<T::AssetNativeLocation>,
+		xcm_rate_limit: Option<Balance>,
+		is_sufficient: bool,
+	) -> Result<Self::AssetId, Self::Error> {
+		let bounded_name = if let Some(name) = name {
+			let bounded_name = Self::to_bounded_name(name.to_vec())?;
+			Some(bounded_name)
+		} else {
+			None
+		};
+
+		let bounded_symbol = if let Some(symbol) = symbol {
+			Some(Self::to_bounded_name(symbol.to_vec())?)
+		} else {
+			None
+		};
+
```

Comment:
might be worth extracting the common logic of these two into a function?
so you get `let bounded_symbol = try_into_bounded(symbol)?;`